### PR TITLE
Deposit insurance cost

### PR DESCRIPTION
**Pull request by aaalata**
Hi!
Please do the Code Review.
Thanks!

### CODE REVIEW

File: src/main/java/pl/edu/agh/iet/katabank/bankproduct/interestpolicy/DailyInterestPolicyWithInsurance.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Override
+    public BigDecimal getYearlyInterestRatePercent() {
+        return this.yearlyInterestRatePercent;
+    }
+
+    @Override
+    public BigDecimal calculateInterest(BigDecimal amount, LocalDate openDate, LocalDate calculationDate, LocalDate plannedTerminationDate) {
+        int depositDuration = getDuration(openDate, plannedTerminationDate.isBefore(calculationDate) ? plannedTerminationDate : calculationDate);
```
Comment (anthonyflea) [inline src/main/java/pl/edu/agh/iet/katabank/bankproduct/interestpolicy/DailyInterestPolicyWithInsurance.java]: Consider extracting "plannedTerminationDate.isBefore(calculationDate) ? plannedTerminationDate : calculationDate" to private method.
Write unit test where calculationDate is after plannedTerminationDate to check if interest is calculated properly.

File: src/test/java/pl/edu/agh/iet/katabank/steps/DepositManagementSteps.java [hunk 2]
```diff
@@ -137,6 +140,24 @@ public DepositManagementSteps() {
             assertThat(account.getBalance()).isEqualByComparingTo(new BigDecimal(balanceWithInterest));
         });
 
+        Given("^there is a customer who is about to open a new deposit of any kind$", () -> {
+            amount = new BigDecimal(1000);
+            account.setBalance(amount);
+        });
+
+        And("^he decided to add the insurance to the deposit$", () -> {
+            interestPolicy = new DailyInterestPolicyWithInsurance(new BigDecimal(10), new BigDecimal(0.05));
```
Comment (anthonyflea) [inline src/test/java/pl/edu/agh/iet/katabank/steps/DepositManagementSteps.java]: Consider changing scenario to "he decided to add the insurance cost (.+)% to the deposit". This way insurance cost could be passed to this method and changed only in deposit_management.feature. Similar as it is done in Then step.

**Review by anthonyflea — APPROVED** [review]
Code review done.

**Comment by aaalata** [comment]
Corrected.

**Pull request merged**